src/Storage/Storage.Management/Blob/UpdateAzureStorageBlobServiceProperties.cs

```diff
@@ -0,0 +1,148 @@
+{
+    using Microsoft.Azure.Commands.Management.Storage.Models;
+    using Microsoft.Azure.Management.Storage;
+    using Microsoft.Azure.Management.Storage.Models;
+    using System;
+    using System.Collections.Generic;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+    /// <summary>
+    /// Modify Azure Storage service properties
+    /// </summary>
+    [Cmdlet("Update", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + StorageBlobServiceProperty, SupportsShouldProcess = true, DefaultParameterSetName = AccountNameParameterSet), OutputType(typeof(PSBlobServiceProperties))]
+    public class UpdateAzStorageBlobServicePropertyCommand : StorageBlobBaseCmdlet
+    {
+        /// <summary>
+        /// AccountName Parameter Set
+        /// </summary>
+        private const string AccountNameParameterSet = "AccountName";
+
+        /// <summary>
+        /// Account object parameter set 
+        /// </summary>
+        private const string AccountObjectParameterSet = "AccountObject";
+
+        /// <summary>
+        /// BlobServiceProperties ResourceId  parameter set 
+        /// </summary>
+        private const string PropertiesResourceIdParameterSet = "BlobServicePropertiesResourceId";
+
+        /// <summary>
+        /// AccountName Parameter Set
+        /// </summary>
+        private const string AccountNamePropertiesObjectParameterSet = "AccountNamePropertiesObject";
+
+        [Parameter(
+          Position = 0,
+          Mandatory = true,
+          HelpMessage = "Resource Group Name.",
+         ParameterSetName = AccountNameParameterSet)]
+        [Parameter(
+          Position = 0,
+          Mandatory = true,
+          HelpMessage = "Resource Group Name.",
+         ParameterSetName = AccountNamePropertiesObjectParameterSet)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            HelpMessage = "Storage Account Name.",
+           ParameterSetName = AccountNameParameterSet)]
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            HelpMessage = "Storage Account Name.",
+           ParameterSetName = AccountNamePropertiesObjectParameterSet)]
+        [ResourceNameCompleter("Microsoft.Storage/storageAccounts", nameof(ResourceGroupName))]
+        [Alias(AccountNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string StorageAccountName { get; set; }
+
+        [Parameter(Mandatory = true,
+            HelpMessage = "Storage account object",
+            ValueFromPipeline = true,
+            ParameterSetName = AccountObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSStorageAccount StorageAccount { get; set; }
```

Comment:
Same comment - this is the only parameter that should take its value from the pipeline

Reply:
As we discussed, the only parameter has ValueFromPipelineByProeprtyName os "-ResourceId" and this is acceptable.
As we discussed, I have change the parameter name in all 4 new cmdlets to "ResourceId", and all will accept both Storage account resourceId and Blob Service ResourceId.